### DISCUSSION TITLE
Is Firebase App Check token retrieval supposed to be this slow?

### DISCUSSION THREAD

**Discussion by synstin** [firebase/flutterfire#13437] [Q&A]
I'm experiencing an issue with Firebase AppCheck token retrieval on iOS. The process is taking between 300ms to 500ms, which seems unusually slow. Here are the key points:

This slow retrieval is only occurring on iOS devices.
On Android, the initial token retrieval takes some time, but subsequent retrievals are much faster (1ms to 10ms).
The slow retrieval on iOS happens even when fetching the same token (which should be cached).
This behavior is consistent across both the simulator and real devices in release mode.
Is this considered normal performance for Firebase AppCheck on iOS? The stark difference between iOS and Android performance is concerning, especially given that cached tokens are also retrieved slowly on iOS.

Has anyone else encountered similar issues or have insights into why this might be happening? Any suggestions for improving the token retrieval speed on iOS would be greatly appreciated.

**Comment by andersonjohnf** [comment]
It's a year later and this is still an issue, how is no one talking about this? AppCheck on iOS is fundamentally broken even on the latest version (0.4.0 as of now). Every single request to getToken() on iOS retrieves a brand-new token (using AppAttest or Debug providers). on Android it is properly cached. Is simply no one actually using AppCheck to protect their custom APIs so no one notices?

**Comment by markwitt1** [comment]
Having the same issues on React native
